server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1082,11 +1058,12 @@ private void addStaleDocs(final List<ParseContext.Document> docs, final IndexWri
         final long versionForIndexing;
         final boolean indexIntoLucene;
         final boolean addStaleOpToLucene;
+        final long seqNoOfNewerVersion; // the seqno of the newer copy of this _uid if exists; otherwise -1
```

Comment:
maybe call this seqNoOfNewerDocIfStale? (trying to avoid the word version as it's confusing)